libs/colbert/ragstack_colbert/colbert_embedding_model.py

```diff
@@ -53,165 +25,87 @@ class ColbertEmbeddingModel(BaseEmbeddingModel):
+            kmeans_niters (Optional[int]): Number of iterations for k-means clustering during quantization.
+            nranks (Optional[int]): Number of ranks (processors) to use for distributed computing; -1 uses all available CPUs/GPUs.
+            query_maxlen (Optional[int]): Maximum length of query tokens for embedding.
+            verbose (Optional[int]): Verbosity level for logging.
+            chunk_batch_size (Optional[int]): The number of chunks to batch during embedding. Defaults to 640.
             **kwargs: Additional keyword arguments for future extensions.
-
-        Note:
-            This initializer also prepares the system for distributed computation if specified and available.
         """
 
-        self.__cuda = torch.cuda.is_available()
-        self.__nranks = reconcile_nranks(nranks)
-        total_visible_gpus = torch.cuda.device_count()
-        logging.info(f"run nranks {self.__nranks}")
-        if (
-            self.__nranks > 1
-            and not dist.is_initialized()
-            and distributed_communication
-        ):
-            logging.info(f"distribution initialization must complete on {nranks} gpus")
-            Distributed(self.__nranks)
-            logging.info("distribution initialization completed")
+        if query_maxlen is None:
+            query_maxlen = -1
 
-        with Run().context(RunConfig(nranks=nranks)):
-            if self.__cuda:
-                torch.cuda.empty_cache()
-            self.colbert_config = ColBERTConfig(
-                doc_maxlen=doc_maxlen,
-                nbits=nbits,
-                kmeans_niters=kmeans_niters,
-                nranks=self.__nranks,
-                checkpoint=checkpoint,
-                query_maxlen=query_maxlen,
-                gpus=total_visible_gpus,
-            )
-        logging.info("creating checkpoint")
-        self.checkpoint = Checkpoint(
-            self.colbert_config.checkpoint,
-            colbert_config=self.colbert_config,
-            verbose=verbose,
-        )
-        self.encoder = CollectionEncoder(
-            config=self.colbert_config, checkpoint=self.checkpoint
+        self._query_maxlen = query_maxlen
+        self._chunk_batch_size = chunk_batch_size
+
+        colbert_config = ColBERTConfig(
+            doc_maxlen=doc_maxlen,
+            nbits=nbits,
+            kmeans_niters=kmeans_niters,
+            nranks=nranks,
+            checkpoint=checkpoint,
         )
-        self.query_tokenizer = QueryTokenizer(self.colbert_config)
-        if self.__cuda:
-            self.checkpoint = self.checkpoint.cuda()
+        self._encoder = TextEncoder(config=colbert_config, verbose=verbose)
 
-    def embed_chunks(
-        self, chunks: List[ChunkData], doc_id: Optional[str] = None
-    ) -> List[EmbeddedChunk]:
+    # implements the Abstract Class Method
+    def embed_texts(self, texts: List[str]) -> List[Embedding]:
```

Comment:
why the doc_id is removed?

Reply:
the embedding model now just handles embedding of text... doesn't need to know about doc_id or metadata.

the doc_id and chunk_id for adding to the database is handled elsewhere